Sources/StreamChatUI/ChatThread/ChatThreadVC.swift

```diff
@@ -66,20 +66,25 @@ open class ChatThreadVC:
         }
 
         messageController.delegate = self
-        let updateMessageControllerMessage: () -> Void = { [messageController, messageComposerVC] in
+
+        let completeSetup: (ChatMessage?) -> Void = { [messageController, messageComposerVC] message in
             if messageComposerVC.content.threadMessage == nil,
                let message = messageController?.message {
                 messageComposerVC.content.threadMessage = message
             }
-        }
-        channelController.synchronize { [messageController] _ in
```

Comment:
AFAIK the channelController.synchronize was needed when opening a Thread from a notification, where the call to the channel was not done. We need to verify if that this use case still works.

Reply:
good point, we should handle this in our demo app (atm we only navigate to the channel and not to the thread)

![image](https://user-images.githubusercontent.com/88735/147064823-7dbbcc28-fe93-41b7-a36e-f7beb64353d5.png)

Reply:
this PR is opening a Pandora box :) 

- our demo app does not open threads when tapping on a message reply
- our docs should have an example `userNotificationCenter:didReceive` that handles both regular messages and replies

Reply:
@nuno-vieira navigating to a thread probably requires adding the channel to the navigation stack as well (otherwise you go back to 🦖)

Reply:
@nuno-vieira @tbarbugli regarding notification about thread reply in the channel that does not exist locally: what if we handle this on the message controller level? 

When `synchronize` is invoked it'll check if there's a channel in the database and if there's no - it'll fetch & save it before fetching a message from the API.

Having a message in the database without the underlying channel makes no sense, right?

Reply:
I think Pavel's solution makes sense 👍

Reply:
`getMessage`, which is called when `MessageController.synchronize` is invoked, already returns `channel` data alongside `message` data. We don't need to call `ChannelController.synchronize`

Reply:
@b-onc Do you test opening a thread from a notification? Just want to make sure we don't have a regression here.

Reply:
@nuno-vieira handling push notification is up to DemoApp. From an SDK perspective we are good, I think.

Reply:
Yes, but the customer will open a thread and the thread needs to correctly load the messages, otherwise, there is a bug on our side. And this bug existed before, so I just wanted to make sure this change did not make a regression regarding that issue.